[Shader Graph] Fix Position Node upgrader

alindmanUnity opened the pull request:
### Purpose of this PR
Why is this PR needed, what hard problem is it solving/fixing? 
Fixed an issue where new Position nodes added to graphs would always be "upgraded", causing the coordinate space drop down to switch from World to Absolute World when entering play mode or restarting the project. 

[public link](https://issuetracker.unity3d.com/product/unity/issues/guid/1184617/)
[internal link](https://fogbugz.unity3d.com/f/cases/1184617/)

---
### Testing status

**Manual Tests**: What did you do?
- [x] Opened test project + Run graphic tests locally
- [x] Built a player
- [x] C# and shader warnings (supress shader cache to see them)
- Other: 
Previous repro behavior: 
- Create a new graph, add a Position node. leave the coordinate space on default World. Save graph.
- Enter and exit playmode to force shader compilation. 
- Open the graph, observe that World has been automatically set to Absolute World. 

Fix behavior: 
- Create a new graph, add a Position node. leave the coordinate space on default World. Save graph.
- Enter and exit playmode to force shader compilation. 
- Open the graph, observe that the coordinate space has not changed.  

**Automated Tests**: What did you setup? (Add a screenshot or the reference image of the test please)

**Yamato**: (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline

Any test projects to go with this to help reviewers?

---
### Comments to reviewers
Notes for the reviewers you have assigned.
This was initially intended to be an upgrader for the position node when Absolute World space was added. Because HDRP uses different values, the intent was that any existing graphs would maintain the same output even though the entry changed. The node version was never set properly and it caused the Position node to always run through the upgrader no matter what. 
Any graphs with the Position node already in use should not see any change in behavior. This only applies to newly created Position nodes after this fix is merged.

marctem commented on com.unity.shadergraph/Editor/Data/Nodes/Input/Geometry/PositionNode.cs:
```diff
@@ -21,6 +21,7 @@ public PositionNode()
         {
             name = "Position";
             precision = Precision.Float;
+            nodeVersion = 1;
```
This should call `GetCompiledNodeVersion();` instead of duplicating the value.

alindmanUnity replied:
added the get compiled node version to AMN constructor per your comment

alindmanUnity commented:
after changing the behavior, new manual testing was: 
1) Create a new graph, add a Position node. leave the coordinate space on default World. Save graph.
2) Enter and exit playmode to force shader compilation.
3) Open the graph, observe that the coordinate space has not changed.
4) Added a new function in the upgrader for Position node to change coordinate space to view on upgrade to version number 2. set compiled node version to 2. 
5) Reloaded graph, observed that existing Position nodes were changed to view space. 
6) added new position, observed that coordinate space of new nodes were not changed.

marctem commented on com.unity.shadergraph/Editor/Data/Nodes/AbstractMaterialNode.cs:
```diff
@@ -28,7 +28,7 @@ abstract class AbstractMaterialNode : ISerializationCallbackReceiver, IGroupItem
 
         [SerializeField]
-        protected int m_NodeVersion;
+        private int m_NodeVersion;
```
Nit: Don't need to say `private`; that's the default.
